dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -261,6 +261,25 @@ cdef class cyDiscreteQuadraticModel:
 
         return biases
 
+    @cython.boundscheck(False)
```

Comment:
If we're not doing these checks here, we need to do some sanity checks in the code body. Otherwise this function can segfault

Reply:
will do a check on the length of all three vectors